### PR TITLE
Blockops Network Grant Application

### PR DESCRIPTION
**Pull request by haroldsphinx**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by haroldsphinx** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by haroldsphinx** [comment]
recheck

**Comment by calvin-puram** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by calvin-puram** [comment]
recheck

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. One quick initial feedback: Your GitHub links, like https://github.com/blockops-sh/api-core-service, currently don't work for me. Also, it would be nice to have more details as part of the milestone specification. For example, what exactly do you mean by "Development of full blown parachain deployment feature"? Could you be as specific as possible here (programming language, functionality, etc.)? The specification of the milestone tables are the requirements of our contracts and we need to be able to evaluate them.

File: applications/blockops_parachain_node_deployment.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+## Development Roadmap :nut_and_bolt:
+
+For this proposal we focus on building and streamlining the process of setting up parachains through a one-click setup dashboard. This would dramatically reduce the time and technical expertise required to bootstrap a parachain network, making the technology more accessible to a broader range of users and accelerating the pace of development within the Polkadot ecosystem.
+
+
+### Overview
+
+- **Total Estimated Duration:** 3 months
+- **Full-Time Equivalent (FTE):**  6 FTE
+- **Total Costs:** 120,000 USD.
```
Comment (semuelle) [inline applications/blockops_parachain_node_deployment.md:230]: The totals don't match the milestone sums.

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the application, @haroldsphinx. I agree with David that the specification of the deliverables need to be more concrete. Please keep in mind that all deliverables should be reproducible and verifiable, and ideally - to avoid misunderstandinds - the specification leaves as little room for interpretation as possible.

**Comment by haroldsphinx** [comment]
Thanks @semuelle and @Noc2 for the swift feedback

Will make the requested changes and fix discrepancy in Estimates as well as updating it with a more detailed Milestone breakdown and revert

**Comment by kukabi** [comment]
As the founder of Helikon, I was never asked for permission for this:

Screenshot 2023-09-05 at 08 58 46

Also it was never discussed in the IBP group chats until it was brought to light by a member.

Please remove it from your proposal.

**Comment by hitchhooker** [comment]
Rotko Networks has no association with this proposal either and even our infrastructure would allow programmatic cloudlike node deploys, there has been absolutely no communication from Blockops regarding what it would take to deploy on our hardware so it feels that all this proposal is to money grab utilizing reputation of existing providers in bad faith with simple frontend without actually providing path to solution.

**Comment by haroldsphinx** [comment]
> As the founder of Helikon, I was never asked for permission for this:
> 
> Screenshot 2023-09-05 at 08 58 46
> Also it was never discussed in the IBP group chats until it was brought to light by a member.
> 
> Please remove it from your proposal.

Done

**Comment by haroldsphinx** [comment]
> Rotko Networks has no association with this proposal either and even our infrastructure would allow programmatic cloudlike node deploys, there has been absolutely no communication from Blockops regarding what it would take to deploy on our hardware so it feels that all this proposal is to money grab utilizing reputation of existing providers in bad faith with simple frontend without actually providing path to solution.


The only reason why we even mentioned IBP or its members in the first place was to demonstrate use case where Parachains can leverage existing Node Providers either Cloud Providers (AWS or GCP) or Dedicated Node Providers (like the ones in IBP). Its simply a way to demonstrate that with our solution Parachains can easily deploy their node infrastructure anywhere. Isn’t that the point of a proposal? to show case the value your solution can provide.

And like we discussed in the channel already. I understand how just mentioning every Dedicated Node Provider in the IBP without talking to them first might be a problem, and I immediately agreed to removing with it which I have gone ahead to remove simply because I didn't even think it will be a problem until it was raised.

What I don’t understand however, is why you (@hitchhooker) are trying to sabotage our proposal because of an “Omission” mistake, that we’re more than willing to correct and have actually did


Also, this idea we have of creating a way for Parachains to deploy their Node to existing Dedicated Node Service Providers has been talked about in our initial proposal to openGov(ref #78), and I as well discussed it in the Polkadot Direction channel, Please find attached the conversations

This was in #PolkadotDirection Channel
https://matrix.to/#/!cINnqxjSEAkadqupkZ:parity.io/$gh6YoBAKLH8r1Gm_-hOhyu1-kf6l6GmIS6Mzw2cDyWU?via=parity.io&via=corepaper.org&via=matrix.org (See the scrrenshot)

This is a quote from the message I sent in the IBP members channel to discuss it:
> _The main reason why I think this will benefit us all especially members of IBP is in-regards to how parachain can choose to deploy to Dedicated Servers in any regions that our IBP providers are, Parachains prefers deploying their infra to Dedicated Servers instead of Public Cloud and I encourage it a lot but its hard and expensive for them to do it alone, but now that we have IBP in different location, imagine parachain choosing to deploy some set of collators on your VPS while you choose to charge per time, this doesn’t just solve the immediate problem, this is also inline to Dr Gav’s talk about the future of Polkadot. ( I attached a screenshot of this)_

Someone even asked us in the channel how we plan on achieving this to which we replied our approach (Please see screenshots attached
Screenshot 2023-09-05 at 9 40 10 AM
Screenshot 2023-09-05 at 9 40 45 AM
)

**Comment by tugytur** [comment]
> As the founder of Helikon, I was never asked for permission for this:
> Screenshot 2023-09-05 at 08 58 46
> 
> Also it was never discussed in the IBP group chats until it was brought to light by a member.
> 
> Please remove it from your proposal.

We (Amforc) would also like to note that we have no association with this grant in any way.

**Comment by haroldsphinx** [comment]
Hello @semuelle @Noc2 

We have attended to all comments and updated our application accordingly

File: applications/blockops_parachain_node_deployment.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+
+- **Who is your target audience?:** Our target audience are dApps developers looking to bootstrap their parachains network and get started quickly and substrate developers substrate developers looking to build and iterate faster as well as test their smart contracts codes against live networks with Beefy built for purpose dedicated machines.
+
+- **What need(s) does your project meet?:** This project will significantly enhance the efficiency and reliability of the development process within the Polkadot ecosystem.
+
+- **Are there any other projects similar to yours in the Substrate / Polkadot / Kusama ecosystem?:** OnFinality has been a successful and proven model for only single node deployment. However Onfinality doesn't offer bootstrappable node deployments for parachains. There is currently no out of the box advanced analytics and proactive monitoring to ensure a production grade incident management of networks and nodes. From that perspective, we are a unique solution that combines both services and offers additional interests to our users.
```
Comment (semuelle) [inline applications/blockops_parachain_node_deployment.md:150]: [Gantree](https://github.com/gantree-io) is not actively maintained currently, but have you looked into what they have built already? Perhaps some of it could be reused. 
Also, are you familiar with what [Tanssi](https://www.tanssi.network/) is doing with regard to infrastructure?

File: applications/blockops_parachain_node_deployment.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+- **Composable Finance**: Setup Networks for Composable Finance for all their chains (Dali, Picasso and Composable)
+- **Polkadot & Kusama**: We have vast experience running RPC and Validator nodes on all Polkadot chains.
+- **Flow Blockchain:** We have experience running Access Nodes and Consensus Nodes on Flow Blockchain
+- We also run Self Managed Nodes (Bitcoin, Ethereum, USDC, USDT etc) for major crypto exchanges in Africa (e.g [Quidax](https://www.quidax.com))
+
+Our team closely monitors the network, stays up to date with upgrades and governance decisions, and adjusts our nodes' configurations accordingly.
+
+Through our experience with Polkadot, we have developed robust strategies to maximize our validator node's performance, optimize staking rewards, and ensure continuous participation in the network's consensus mechanism.
+
+### Team Code Repos
```
Comment (semuelle) [inline applications/blockops_parachain_node_deployment.md:191]: Since these repos are largely empty, does (anyone on) your team have experience with writing and maintaining open source software that you could share?

**Review by semuelle — CHANGES_REQUESTED** [review]
Thank you for the application and sorry for the late reply, @haroldsphinx. I'll try to be more responsive from now on. Below are a couple of high-level questions and change requests.

File: applications/blockops_parachain_node_deployment.md [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+| **0e** | Article | Using our communication channels, we will publish articles that explains the high-level architecture of our monitoring infrastructure and how users can set them up on their environment 
+| 1. | block-agent Architecture	| Design and develop the architecture of the block-agent, ensuring it's modular and scalable. | The agent will efficiently collect vital metrics such as CPU usage, memory consumption, network latency, and other relevant node performance indicators. | 
+| 2. | Development of One Click Parachain Deployment Feature on Dedicated server | Development of Dedicated Node Provider Marketplace |
+| 3. | Improve UX Analytics Dashboard | Detailed reports on node performance, including usage statistics, latency, and uptime. | Log Dashbooard | Ability to filter logs by different Log Levels | Setup Dashboard: We will setup both standalone and customize dashboards that will log and give insight on operations perform by the different users
+| 4. | Package blocks-cli tool for deployment Docker, MacOS and Ubuntu |  Successful security and performance audit of the platform | 
+
```
[the remark this answers is not in the document]
Comment (semuelle) [reply applications/blockops_parachain_node_deployment.md:268]: (the empty lines are breaking the markdown rendering)

File: applications/blockops_parachain_node_deployment.md (hunk 3 above)
Comment (haroldsphinx) [reply applications/blockops_parachain_node_deployment.md:191]: > Since these repos are largely empty, does (anyone on) your team have experience with writing and maintaining open source software that you could share?

Yes, 3 of our Infrastructure members have this experience but mostly on Contributing and not Maintaining

File: applications/blockops_parachain_node_deployment.md (hunk 2 above)
Comment (haroldsphinx) [reply applications/blockops_parachain_node_deployment.md:150]: Thanks for sharing the Gantree reference, check them out and we can draw learnings from their implantation.

The common gooal with these two is that we are all trying to solve the same problems in different ways and so far we are closer with a Simpler working solution.

**Comment by haroldsphinx** [comment]
Hello @semuelle

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Hi @haroldsphinx thanks for the application. However, we don't offer DOT as a payment method. Please change to one of these [payment options](https://github.com/w3f/Grants-Program#pencil-process). Thanks.

**Comment by keeganquigley** [comment]
@haroldsphinx also, I'm curious as to why you raised the price to 90k?

**Comment by haroldsphinx** [comment]
> Hi @haroldsphinx thanks for the application. However, we don't offer DOT as a payment method. Please change to one of these [payment options](https://github.com/w3f/Grants-Program#pencil-process). Thanks.

Will update accordingly.

**Comment by haroldsphinx** [comment]
> @haroldsphinx also, I'm curious as to why you raised the price to 90k?

Our initial calculated cost base don FTE can be found [here](https://docs.google.com/document/d/15et4Lm9T1_f74GLTgCboaMNrO-QjNPSybv5qJq84P4Q/edit#heading=h.kz6aqfpbquoy) 

The initial price we set right from our first proposal was $160k and this was based on actual calculation FTE's we will need to achieve this, but based on feedback and other factors, we decided to go extra-lean by removing some redundant roles that and also reducing the number of workforce.

I initially reduced the cost to the lowest possible amount that will get us through. However my team has been working non-stop on the project since the first time we submitted, and they are really burned out and its affecting the pace of work, and I re-calculated on how much extra manpower will make us work at a decent rate while achieving the same goal under the same timeline, and that was why I updated it.

File: applications/blockops_parachain_node_deployment.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+| **0e** | Article | Using our communication channels, we will publish articles that explains the high-level architecture of our monitoring infrastructure and how users can set them up on their environment
+| 1. | block-agent Architecture	| Design and develop the architecture of the block-agent, ensuring it's modular and scalable. | The agent will efficiently collect vital metrics such as CPU usage, memory consumption, network latency, and other relevant node performance indicators. |
+| 2. | Development of One Click Parachain Deployment Feature on Dedicated server | Development of Dedicated Node Provider Marketplace |
+| 3. | Improve UX Analytics Dashboard | Detailed reports on node performance, including usage statistics, latency, and uptime. | Log Dashbooard | Ability to filter logs by different Log Levels | Setup Dashboard: We will setup both standalone and customize dashboards that will log and give insight on operations perform by the different users
+| 4. | Package blocks-cli tool for deployment Docker, MacOS and Ubuntu |  Successful security and performance audit of the platform |
```
Comment (semuelle) [inline applications/blockops_parachain_node_deployment.md:266]: Audits are specifically excluded from the grants program scope. Please see our [FAQ](https://grants.web3.foundation/docs/faq#what-activitiespositions-do-you-fund) for more details.

File: applications/blockops_parachain_node_deployment.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+- **Costs:** 45,000 USD
+
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) set up parachains through a one-click setup. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0e. | Article | Using our communication channels, we will publish an article that explains the high-level overview of automation as part of the grant, followed by a set of extensive examples. |
+| 1. | Completion of the initial development phase of the one-click feature. | Development of One Click Parachain Deployment Feature on GCP, AWS & Digital Ocean | Alpha testing | User Centric Sleek Interface Design |
```
Comment (semuelle) [inline applications/blockops_parachain_node_deployment.md:241]: The `|`s break the markdown rendering and everything after doesn't show up. Please also check the other rows in both tables.

```suggestion
| 1. | Completion of the initial development phase of the one-click feature. | Development of One Click Parachain Deployment Feature on GCP, AWS & Digital Ocean; Alpha testing; User Centric Sleek Interface Design |
```

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks again for the application, @haroldsphinx, and sorry for the late reply. 

I left a couple of comments and questions below. Generally, the scope of this application is quite large and the description/specifications relatively high level. It would be nice to get a bit more details on the architecture. For example, D1.3 mentions key rotation and client upgrades, but not what work is necessary to accomplish this. It might make sense to apply for a smaller scope initially. Anyway, once you have fixed the issues I raised below, I will share the application with the rest of the committee, who might have other ideas.

File: applications/blockops_parachain_node_deployment.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+
+
+- **Who is your target audience?:** Our target audience are dApps developers looking to bootstrap their parachains network and get started quickly and substrate developers substrate developers looking to build and iterate faster as well as test their smart contracts codes against live networks with Beefy built for purpose dedicated machines.
+
+- **What need(s) does your project meet?:** This project will significantly enhance the efficiency and reliability of the development process within the Polkadot ecosystem.
+
+- **Are there any other projects similar to yours in the Substrate / Polkadot / Kusama ecosystem?:** OnFinality has been a successful and proven model for only single node deployment. However Onfinality doesn't offer bootstrappable node deployments for parachains. There is currently no out of the box advanced analytics and proactive monitoring to ensure a production grade incident management of networks and nodes. From that perspective, we are a unique solution that combines both services and offers additional interests to our users.
```
[the remark this answers is not in the document]
Comment (haroldsphinx) [reply applications/blockops_parachain_node_deployment.md:150]: Yes they do, but its not open source and doesn't give parachains the flexibility of bootstrapping a parachain at once, and its also subject to their review, all of which doesn't sit well with Parachains in terms of Trust

File: applications/blockops_parachain_node_deployment.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+
+We have developed a one-click deployment feature for Parachains, Node Operators and Networks looking to setup robust Infrastructure on Polkadot.
+
+This platform provides substrate developers with an effortless solution to easily bootstrap their entire network and deploy their Parachain Networks on a public cloud or Dedicated Servers situated within the geographic regions where the Node Providers willing to be provde Dedicated Node Service resides. This ensures that developers and enterprises can build faster and manage parachains easily.
+
+The user interface and infrastructure set up for this project are largely complete and ready. To get a glimpse of the minimum viable product (MVP), you can the [website](https://beta-app.blockops.network/login).
+
+**The System Overview**
+
+The project is composed of three primary open-source modules designed to support node infrastructures:
```
[the remark this answers is not in the document]
Comment (haroldsphinx) [reply applications/blockops_parachain_node_deployment.md:84]: Thanks, I have added a section to highlight the current status of development for each roadmap

**Review by semuelle — COMMENTED** [review]
Thank you for the updates, @haroldsphinx. I am somewhat skeptical that the committee is willing to spend 90k on a deployment tooling project in one fell swoop, but I am going to share it with the rest of the committee regardless.

**Comment by haroldsphinx** [comment]
> Thank you for the updates, @haroldsphinx. I am somewhat skeptical that the committee is willing to spend 90k on a deployment tooling project in one fell swoop, but I am going to share it with the rest of the committee regardless.

Please can you advice on the best way to structure this? This is the reason why we break it down into Milestone #1 and #2 and I didn't want to be re-adjusting costs too often but I'm willing to adjust this to what will work because what is important to us right now is that the Project is accepted, so it gives us the confidence that we're spending effort building something that is really needed.

 We are willing to adjust the pricing to what is feasible

**Comment by haroldsphinx** [comment]
Made the following updates:

- [ ] Added USDC Address
- [ ] Reduced the total pricing back to $60k
- [ ] Updated Development Deliverables to Include a Launcher as web believe this will be the best way to Achieve True Decentralization and Sovereignty  by giving users the autonomy to use this tool on any platform to deploy to any environment
- [ ] Added the link to the open sourced codes to also showcase progress and continuous workdone even as we are waiting on the review process

Cc: @semuelle @keeganquigley @Noc2

**Comment by haroldsphinx** [comment]
I marked Old conversation resolved, to clean up the comments a bit and also as a way to nudge reviewers 🙏🏾

**Comment by haroldsphinx** [comment]
> @haroldsphinx thanks for the adjustments and the price reduction. Usually we ask teams to start with a L1 or L2 grant and if the deliveries turn out to be of high quality we consider follow-up grants. For us, that's a way to mitigate the risk to spend a lot of money with little in return. Would you consider to strip M2 and apply for M1 only?
> 
> Apart from that, I've seen that your team members currently work for different companies, such as Dapper Labs, Exodus, BlockOps Network, Kora and Composable Finance. At the same time, you plan to allocate 6 FTE for 6 months. So I was wondering if they are all going to resign from their jobs to work full-time on that project. Because I can't imagine anybody to work 200% for extended periods of time (in this case, half a year).



> @haroldsphinx thanks for the adjustments and the price reduction. Usually we ask teams to start with a L1 or L2 grant and if the deliveries turn out to be of high quality we consider follow-up grants. For us, that's a way to mitigate the risk to spend a lot of money with little in return. Would you consider to strip M2 and apply for M1 only?
> 
> Apart from that, I've seen that your team members currently work for different companies, such as Dapper Labs, Exodus, BlockOps Network, Kora and Composable Finance. At the same time, you plan to allocate 6 FTE for 6 months. So I was wondering if they are all going to resign from their jobs to work full-time on that project. Because I can't imagine anybody to work 200% for extended periods of time (in this case, half a year).

Thank you for the feedback @takahser 

Regarding M2 and Grant Phases:
In light of your suggestion, I have updated our application to focus solely on M1 for now. We’re optimistic that after the successful delivery of M1, we would be in a good position to apply for follow-up grants.

Regarding Team Members’ Current Employment Status:

To provide clarity on our team’s dedication and commitment, here are the details of each member’s current work status:
Adedayo Akinpelu: Worked with Composable Finance followed by Dapperlabs. However, I left Dapperlabs in July this year, which has afforded me the time to concentrate fully on Blockops. FTE
Marcelo Mendoca: Worked with Dapperlabs but left recently, Partime
Franklin Okpako: He is presently employed at Exodus. Partime
Oluwatimilehin Adesoji: He is currently with Korapay. Partime
John Etokana: Works with Composable Finance but left recently…Partime
Calvin: Full-time with Blockops Network.

Also, the way I’m calculating the FTE is in terms of number of hours worked, Those team members who are not engaged in any other full-time job are wholeheartedly investing additional hours into this project to ensure its success.

**Comment by takahser** [comment]
@haroldsphinx thanks for the adjustments and the reply. Regarding the allocations, how large are the part-time allocations of the other team members? Can you list the allocations for this project for each of your team members?

> To provide clarity on our team’s dedication and commitment, here are the details of each member’s current work status:
> Adedayo Akinpelu: Worked with Composable Finance followed by Dapperlabs. However, I left Dapperlabs in July this year, which has afforded me the time to concentrate fully on Blockops. FTE
> Marcelo Mendoca: Worked with Dapperlabs but left recently, Partime
> Franklin Okpako: He is presently employed at Exodus. Partime
> Oluwatimilehin Adesoji: He is currently with Korapay. Partime
> John Etokana: Works with Composable Finance but left recently…Partime
> Calvin: Full-time with Blockops Network.

**Comment by haroldsphinx** [comment]
> @haroldsphinx thanks for the adjustments and the reply. Regarding the allocations, how large are the part-time allocations of the other team members? Can you list the allocations for this project for each of your team members?
> 
> > To provide clarity on our team’s dedication and commitment, here are the details of each member’s current work status:
> > Adedayo Akinpelu: Worked with Composable Finance followed by Dapperlabs. However, I left Dapperlabs in July this year, which has afforded me the time to concentrate fully on Blockops. FTE
> > Marcelo Mendoca: Worked with Dapperlabs but left recently, Partime
> > Franklin Okpako: He is presently employed at Exodus. Partime
> > Oluwatimilehin Adesoji: He is currently with Korapay. Partime
> > John Etokana: Works with Composable Finance but left recently…Partime
> > Calvin: Full-time with Blockops Network.


Work Hours Allocation for Parttime Team Members 
Franklin Okpako: 40hrs weekly (currently works equivalent of FTE, he's able to combine the two becuase of his company's timezone)
Marcelo Mendoca: 20hrs weekly including weekends
Oluwatimilehin Adesoji: 20hrs weekly including weekends
John Etokana:  20hrs weekly including weekends

**Comment by takahser** [comment]
@haroldsphinx thanks for the update. It's a bit unfortunate, that you weren't able to deliver the correct FTE from the beginning. Also, I'm unclear on why you didn't update the amounts although the FTE were reduced dramatically. Combined with the complains of different parachains in earlier comments [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1705996330) and [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1706095003) this doesn't shed a very good light on the project - at least that's my personal opinion. To be honest, I'm not very much in favor of accepting this proposal and will withhold my vote for now. However, I'll ping the rest of the committee to have a look again in order to conclude this proposal soon.

**Comment by obol-platform** [comment]
> @haroldsphinx thanks for the update. It's a bit unfortunate, that you weren't able to deliver the correct FTE from the beginning. Also, I'm unclear on why you didn't update the amounts although the FTE were reduced dramatically. Combined with the complains of different parachains in earlier comments [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1705996330) and [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1706095003) this doesn't shed a very good light on the project - at least that's my personal opinion. To be honest, I'm not very much in favor of accepting this proposal and will withhold my vote for now. However, I'll ping the rest of the committee to have a look again in order to conclude this proposal soon.

- I updated the FTE details in the interest of accuracy and transparency. While each team member is involved in other full-time roles, they remain deeply committed to this project and are working on a Full-time capacity on Blockops at the same time. However, based on your previous comment, there seems to be skepticism regarding the feasibility of such a commitment alongside other full-time engagements which is why I updated to something more verifiable and trackable and each member of the team can be held accountable on.

- I opted not to adjust the price even with the FTE reduction because, frankly, $30k already significantly undervalues our efforts. Our primary aim isn't financial; we're passionate about ensuring this project reaches its potential. By any standard, $30k for 4.5 FTEs over three months is considerably below market rate. However, given the challenges we've been facing, we believe it's crucial to demonstrate our capabilities by successfully delivering on our milestones before considering financial aspects in depth. So much that in my mail to the foundation, I stated that my team and I are happy to not receive any form of funding until the Milestones are completed delivered as a show of good faith and our confidence in actually delivering this solution to the ecosystem.

- It's disheartening when it appears that applications can be criticized without the Web3 Foundation extending the courtesy of the benefit of the doubt to applicants or providing fair hearing and hence judging the applications by these innuendos. I have communicated our position and intentions in a detailed email to [grants@web3.foundation] and provided evidence to address all concerns. I did that both as a reply to the Issue Comments in question, but also as a detailed mail to the foundation itself. Unfortunately, I haven’t received a response acknowledging receipts or asking further clarifying questions to shed light on the matter. [See attached screenshot for evidence]

- Also, it is important to note that during the course of this application, 2 parachains has used our beta version, One is currently using our product for their testate phase, and the other is currently in a test phase (one of them even gave us a review and feedbacks on things to improve on ) which is part of the feedbacks we are working on and fixing bugs as well, as of which can be seen in our repository

IMG_2261

**In Conclusion**
I appreciate all the efforts the reviewer has been putting into this application, and I will also like to request that the committee take this review on the focus on the Value Add to the Ecosystem and be a little lenient in Application related errors in our PR, and guide us into fixing this instead.


Screenshot 2023-10-29 at 12 13 43 PM

**Comment by haroldsphinx** [comment]
> @haroldsphinx thanks for the update. It's a bit unfortunate, that you weren't able to deliver the correct FTE from the beginning. Also, I'm unclear on why you didn't update the amounts although the FTE were reduced dramatically. Combined with the complains of different parachains in earlier comments [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1705996330) and [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1706095003) this doesn't shed a very good light on the project - at least that's my personal opinion. To be honest, I'm not very much in favor of accepting this proposal and will withhold my vote for now. However, I'll ping the rest of the committee to have a look again in order to conclude this proposal soon.

> @haroldsphinx thanks for the update. It's a bit unfortunate, that you weren't able to deliver the correct FTE from the beginning. Also, I'm unclear on why you didn't update the amounts although the FTE were reduced dramatically. Combined with the complains of different parachains in earlier comments [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1705996330) and [here](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1706095003) this doesn't shed a very good light on the project - at least that's my personal opinion. To be honest, I'm not very much in favor of accepting this proposal and will withhold my vote for now. However, I'll ping the rest of the committee to have a look again in order to conclude this proposal soon.

- I updated the FTE details in the interest of accuracy and transparency. While each team member is involved in other full-time roles, they remain deeply committed to this project and are working on a Full-time capacity on Blockops at the same time. However, based on your previous comment, there seems to be skepticism regarding the feasibility of such a commitment alongside other full-time engagements which is why I updated to something more verifiable and trackable and each member of the team can be held accountable on.

- I opted not to adjust the price even with the FTE reduction because, frankly, $30k already significantly undervalues our efforts. Our primary aim isn't financial; we're passionate about ensuring this project reaches its potential. By any standard, $30k for 4.5 FTEs over three months is considerably below market rate. However, given the challenges we've been facing, we believe it's crucial to demonstrate our capabilities by successfully delivering on our milestones before considering financial aspects in depth. So much that in my mail to the foundation, I stated that my team and I are happy to not receive any form of funding until the Milestones are completed delivered as a show of good faith and our confidence in actually delivering this solution to the ecosystem.

- It's disheartening when it appears that applications can be criticized without the Web3 Foundation extending the courtesy of the benefit of the doubt to applicants or providing fair hearing and hence judging the applications by these innuendos. I have communicated our position and intentions in a detailed email to [grants@web3.foundation] and provided evidence to address all concerns. I did that both as a reply to the Issue Comments in question, but also as a detailed mail to the foundation itself. Unfortunately, I haven’t received a response acknowledging receipts or asking further clarifying questions to shed light on the matter. [See attached screenshot for evidence]

- Also, it is important to note that during the course of this application, 2 parachains has used our beta version, One is currently using our product for their testate phase, and the other is currently in a test phase (one of them even gave us a review and feedbacks on things to improve on ) which is part of the feedbacks we are working on and fixing bugs as well, as of which can be seen in our repository

IMG_2261

**In Conclusion**
I appreciate all the efforts the reviewer has been putting into this application, and I will also like to request that the committee take this review on the focus on the Value Add to the Ecosystem and be a little lenient in Application related errors in our PR, and guide us into fixing this instead.


Screenshot 2023-10-29 at 12 13 43 PM

**Comment by takahser** [comment]
> I updated the FTE details in the interest of accuracy and transparency. While each team member is involved in other full-time roles, they remain deeply committed to this project and are working on a Full-time capacity on Blockops at the same time. However, based on your previous comment, there seems to be skepticism regarding the feasibility of such a commitment alongside other full-time engagements which is why I updated to something more verifiable and trackable and each member of the team can be held accountable on.

Thanks, I appreciate the adjustments. However, I don't understand why you didn't state the correct FTE from the beginning.

> my team and I are happy to not receive any form of funding until the Milestones are completed delivered as a show of good faith and our confidence in actually delivering this solution to the ecosystem.

A completed and approved delivery is actually the pre-condition for payment in our program, as you can read [here](https://github.com/w3f/Grants-Program/tree/f0bc40076637fcea398c2557be1675aec9e5d6bc#3-milestone-delivery-and-payment).

> Also, it is important to note that during the course of this application, 2 parachains has used our beta version, One is currently using our product for their testate phase, and the other is currently in a test phase (one of them even gave us a review and feedbacks on things to improve on ) which is part of the feedbacks we are working on and fixing bugs as well, as of which can be seen in our repository.

That's great, I'm happy for you! I wish you succeed with your project.

Regarding the email, we're recently receiving a lot of messages and although we try to reply to all of them, there are instances where this doesn't happen. In your case, I'd like to add that you didn't ask for any specific question or feedback, but nonetheless I agree that it'd have been better to confirm the receipt of that message.

**Comment by takahser** [comment]
@haroldsphinx I saw you've requested a review from me. However, I already [commented](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1785500673) recently on it and I don't see any changes or follow-up comments that were created ever since. Let me know if I'm missing something.

**Comment by haroldsphinx** [comment]
> @haroldsphinx I saw you've requested a review from me. However, I already [commented](https://github.com/w3f/Grants-Program/pull/1946#issuecomment-1785500673) recently on it and I don't see any changes or follow-up comments that were created ever since. Let me know if I'm missing something.

Oh, My bad...In response to your comments

> Thanks, I appreciate the adjustments. However, I don't understand why you didn't state the correct FTE from the beginning.

Since the course of our application, we have made several changes to this application based on feedbacks, for-example, we’ve changed the cost from $120k to $60k to $90k and  back to $60k again, We adjusted our deliverables severally to follow inline with progress made and as well as added to our deliverables, all these are done mostly based on feedbacks gotten from reviewers to improve our application, when you brought up the “FTE and working 200%“, I went back to read other projects and realize that we needed to be more specific around the number of hours each member of the team is going to be putting into the project, and based on that I re-calculated and made adjustments accordingly

**Comment by haroldsphinx** [comment]
To the reviewers @takahser @semuelle @Noc2 @keeganquigley 

I again urge the reviewers to consider our application on evaluating our application based on the intrinsic value of the project we’re creating and its potential benefits to the ecosystem. Any other concerns can be addressed and adjusted according to the guidelines set by the Web3 Foundation.

I'm aware that the team is handling  lot of reviews and some are even OOO, Nonetheless,  I would appreciate an update on when we can anticipate the team's final decision regarding the proposal since we've submitted since September 1st. This information is crucial for us to progress collectively 🙏🏾.
